lib/kitchen/directions/bake_chapter_key_concepts/v1.rb

```diff
@@ -16,16 +18,18 @@ def bake(chapter:, metadata_source:, append_to:)
         title = Kitchen::Directions::EocSectionTitleLinkSnippet.v1(page: page)
         key_concepts.each do |key_concept|
           key_concept.prepend(child: title)
-          key_concept&.cut(to: key_concepts_clipboard)
+          key_concept.wrap("<div class='os-section-area'>")
+          page.search('.os-section-area').first.cut(to: key_concepts_clipboard)
         end
-        @key_concepts.push(key_concepts_clipboard.paste)
-        key_concepts_clipboard.clear
       end
 
+      @content = "<div class=\"os-key-concepts\"> #{key_concepts_clipboard.paste} </div>"
+
       append_to_element = append_to || chapter
-      @title_tag = append_to ? 'h3' : 'h2'
+      @in_composite_chapter = append_to_element[:'data-type'] == 'composite-chapter'
```

Comment:
so that we keep the `data-type` knowledge all in one place, can you add an `is?(type)` method to `ElementBase`?  Has a few steps

```ruby
class ElementBase
  def self.descendant(type)
    @types_to_descendants ||=
      descendants.each_with_object({}) do |descendant, hash|
        hash[descendant.short_type] = descendant
      end

    @types_to_descendants[type]
  end

  def self.descendant!(type)
    descendant(type) || raise("Unknown ElementBase descendant type '#{type}'")
  end

  def is?(type)
    self.class.descendant!(type).is_the_element_class_for?(raw)
  end
end
```

`is_the_element_class_for?` already exists in each element class.  Then add a class method `short_type` to each element class, e.g. changing 

```ruby
class CompositeChapterElement < ElementBase
  def initialize(node:, document: nil)
      super(node: node,
            document: document,
            enumerator_class: ElementEnumerator,
            short_type: :composite_chapter)
    end
```

to

```ruby
class CompositeChapterElement < ElementBase
  def self.short_type
    :composite_chapter
  end

  def initialize(node:, document: nil)
      super(node: node,
            document: document,
            enumerator_class: ElementEnumerator,
            short_type: self.class.short_type)
    end
```

And then this line becomes

```ruby
@in_composite_chapter = append_to_element.is?(:composite_chapter)
```